Sources/StreamChat/Models/Attachments/ChatMessageImageAttachment.swift

```diff
@@ -42,19 +50,16 @@ extension ImageAttachmentPayload: Encodable {
 extension ImageAttachmentPayload: Decodable {
     public init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: AttachmentCodingKeys.self)
-
-        guard
-            let imageURL = (
-                try container.decodeIfPresent(String.self, forKey: .image)
-                    ?? container.decodeIfPresent(String.self, forKey: .imageURL)
-                    ?? container.decodeIfPresent(String.self, forKey: .assetURL)
-            )?.attachmentFixedURL
-        else { throw ClientError.AttachmentDecoding() }
         
-        let imagePreviewURL = try container
-            .decodeIfPresent(String.self, forKey: .thumbURL)?
-            .attachmentFixedURL
+        guard case let .dictionary(payload) = try RawJSON(from: decoder) else {
+            throw ClientError.AttachmentDecoding("Failed to parse image payload extra data.")
+        }
 
+        let imageURL = try
+            container.decodeIfPresent(URL.self, forKey: .image) ??
+            container.decodeIfPresent(URL.self, forKey: .imageURL) ??
+            container.decode(URL.self, forKey: .assetURL)
```

Comment:
@tbarbugli are we doing the right thing when using a fallback chain for image URL and title?